File: ainpc-api/src/main/java/ro/ainpc/addons/AINPCAddon.java
package ro.ainpc.addons;

import ro.ainpc.api.AINPCPlatformApi;

public interface AINPCAddon {

    AddonDescriptor getDescriptor();

    default void onLoad(AINPCPlatformApi api) {
    }

    default void onEnable(AINPCPlatformApi api) {
    }

    default void onDisable(AINPCPlatformApi api) {
    }
}

File: ainpc-api/bin/main/ro/ainpc/addons/AINPCAddon.kt
package ro.ainpc.addons

import ro.ainpc.api.AINPCPlatformApi

interface AINPCAddon {
    fun getDescriptor(): AddonDescriptor

    fun onLoad(api: AINPCPlatformApi) {}

    fun onEnable(api: AINPCPlatformApi) {}

    fun onDisable(api: AINPCPlatformApi) {}
}
